src/App.tsx

```diff
@@ -1,26 +1,266 @@
+                    <div className="button-box-top">
+                        <div className="button-col">
+                            <NormalButton/>
+                            <CornerButton/>
+                            <UndoButton/>
+                            <RedoButton/>
+                        </div>
+                        <Numpad
+                        onClickNum={(i: SquareValue) => {this.normalMarkSelectedSquares(i); }}
+                        onClickDel={() => {this.deleteSelectedSquares(); }}
+                        />
+                    </div>
+                    <div className="button-box-bot">
+                        <Button variant="contained" color="primary">
+                          R E S T A R T
+                        </Button>
+                        <Button variant="contained" color="secondary">
+                          C H E C K
+                        </Button>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    public renderSquares() {
+        const list = [];
+        for (let i = 0; i < 81; i++) {
+            list.push(this.renderSquare(i));
+        }
+        return list;
+    }
+
+    public renderSquare(i: SquareAddress) {
+        const isHighlighted = this.state.highlights.includes(i);
+        const isContradicting = this.state.contradicts.includes(i);
+        const marks = i % 9 + 1 == 7 ? [i % 1 as SquareValue, i % 5 + 1 as SquareValue, i % 3 + 1 as SquareValue, i % 8 + 1 as SquareValue] : [];
+        return <Square value={this.state.values[i]}
+        isPermanent={this.state.permanentValues.includes(i)} isHighlighted = {isHighlighted} isContradicting={isContradicting}
+        markings={marks} onClick={() => { this.toggleSelectSquare(i); }}/>;
+    }
+}
+
+interface SquareProps {
+    isPermanent: boolean;
+    isHighlighted: boolean;
+    isContradicting: boolean;
+    value: SquareValue | null;
+    markings: SquareValue[];
+    onClick: () => void;
+}
+
+type SquareValue = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 ;
+
+class Square extends React.Component<SquareProps, {}> {
+    public getMarks(): boolean[] {
+        const lst: boolean[] = [];
+        for (let i = 0; i < 9; i++) {
+            lst[i] = this.props.markings.includes(i as SquareValue);
+        }
+        return lst;
+    }
+    public render() {
+        const light: string = this.props.isHighlighted ? "highlight" : (this.props.isContradicting ? "contradict" : "");
+        const marks: boolean[] = this.getMarks();
+        const displayMarkings: (boolean | 0) = // determine if markings or value should be displayed
+            this.props.markings && this.props.markings.length &&
+            !this.props.isPermanent && !this.props.value;
+        if (displayMarkings) {
+            return(
+                <div className={"square " + light} onClick={this.props.onClick}>
+                    <div className="mark">{marks[1] ? 1 : null}</div>
```

Comment:
Replace these with a loop

Reply:
Use the `.map` method on Arrays:
https://alligator.io/react/rendering-arrays-in-react/